common/src/main/scala/akka/contrib/persistence/mongodb/MongoReadJournal.scala

```diff
@@ -22,25 +25,50 @@ class MongoReadJournal(system: ExtendedActorSystem, config: Config) extends Read
   override def javadslReadJournal(): javadsl.ReadJournal = new JavaDslMongoReadJournal(new ScalaDslMongoReadJournal(impl))
 }
 
-class ScalaDslMongoReadJournal(impl: MongoPersistenceReadJournallingApi) extends scaladsl.ReadJournal with CurrentPersistenceIdsQuery with CurrentEventsByPersistenceIdQuery {
+class ScalaDslMongoReadJournal(impl: MongoPersistenceReadJournallingApi)
+    extends scaladsl.ReadJournal
+    with CurrentPersistenceIdsQuery
+    with CurrentEventsByPersistenceIdQuery
+    with EventsByPersistenceIdQuery{
 
-  def allEvents(): Source[EventEnvelope,Unit] =
-    Source.actorPublisher[EventEnvelope](impl.allEvents)
+  def currentAllEvents(): Source[EventEnvelope,Unit] =
+    Source.actorPublisher[EventEnvelope](impl.currentAllEvents)
       .mapMaterializedValue(_ => ())
 
   override def currentPersistenceIds(): Source[String, Unit] =
-    Source.actorPublisher[String](impl.allPersistenceIds)
+    Source.actorPublisher[String](impl.currentPersistenceIds)
       .mapMaterializedValue(_ => ())
 
   override def currentEventsByPersistenceId(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long): Source[EventEnvelope, Unit] = {
     require(persistenceId != null, "PersistenceId must not be null")
-    Source.actorPublisher[EventEnvelope](impl.eventsByPersistenceId(persistenceId,fromSequenceNr,toSequenceNr))
-      .mapMaterializedValue(_ => ())
+    Source.actorPublisher[Event](impl.currentEventsByPersistenceId(persistenceId, fromSequenceNr, toSequenceNr))
+      .via(Flow[Event].transform(() => new EventEnvelopeConverter)).mapMaterializedValue(_ => ())
+  }
+
+  override def eventsByPersistenceId(persistenceId: String, fromSequenceNr: Long, toSequenceNr: Long): Source[EventEnvelope, Unit] = {
```

Comment:
This stream turned out a lot simpler than I expected.  That's a good thing :smile: